bindings/node/src/Version.cpp

```diff
@@ -0,0 +1,140 @@
+#include <arbiter/Version.h>
+#include "Version.h"
+
+namespace ArbiterNodeBindings {
+
+using v8::Context;
+using v8::Function;
+using v8::FunctionCallbackInfo;
+using v8::FunctionTemplate;
+using v8::Isolate;
+using v8::Just;
+using v8::Local;
+using v8::Maybe;
+using v8::Nothing;
+using v8::Number;
+using v8::Object;
+using v8::Persistent;
+using v8::String;
+using v8::Value;
+
+Persistent<Function> SemanticVersion::constructor;
+
+SemanticVersion::SemanticVersion(unsigned major, unsigned minor, unsigned patch, Maybe<char *> prereleaseVersion, Maybe<char *> buildMetadata) {
+  _semanticVersion = ArbiterCreateSemanticVersion(
+    major,
+    minor,
+    patch,
+    (prereleaseVersion.IsJust() ? prereleaseVersion.FromJust() : NULL),
+    (buildMetadata.IsJust() ? buildMetadata.FromJust() : NULL)
+  );
+}
+
+SemanticVersion::SemanticVersion(const char *string) {
+  _semanticVersion = ArbiterCreateSemanticVersionFromString(string);
+}
+
+SemanticVersion::~SemanticVersion() {
+}
+
+void SemanticVersion::Init(Isolate* isolate) {
+  // Prepare constructor template
+  Local<FunctionTemplate> tpl = FunctionTemplate::New(isolate, New);
+  tpl->SetClassName(String::NewFromUtf8(isolate, "SemanticVersion"));
+  tpl->InstanceTemplate()->SetInternalFieldCount(1);
+
+  // Prototype
+  NODE_SET_PROTOTYPE_METHOD(tpl, "getMajorVersion", GetMajorVersion);
+  NODE_SET_PROTOTYPE_METHOD(tpl, "getMinorVersion", GetMinorVersion);
+  NODE_SET_PROTOTYPE_METHOD(tpl, "getPatchVersion", GetPatchVersion);
+  NODE_SET_PROTOTYPE_METHOD(tpl, "getPrereleaseVersion", GetPrereleaseVersion);
+  NODE_SET_PROTOTYPE_METHOD(tpl, "getBuildMetadata", GetBuildMetadata);
+
+  constructor.Reset(isolate, tpl->GetFunction());
+}
+
+void SemanticVersion::New(const FunctionCallbackInfo<Value>& args) {
+  SemanticVersion *obj;
+
+  if (args.Length() > 1) {
+    unsigned major = args[0]->Uint32Value();
+    unsigned minor = args[1]->Uint32Value();
+    unsigned patch = args[2]->Uint32Value();
+    Maybe<char *> prereleaseVersion = (args[3]->IsNull() || args[3]->IsUndefined()) ? Nothing<char *>() : Just(*String::Utf8Value(args[3]->ToString()));
+    Maybe<char *> buildMetadata = (args[4]->IsNull() || args[4]->IsUndefined()) ? Nothing<char *>() : Just(*String::Utf8Value(args[4]->ToString()));
+    obj = new SemanticVersion(major, minor, patch, prereleaseVersion, buildMetadata);
+  } else {
+    const char *string = *String::Utf8Value(args[0]->ToString());
+    obj = new SemanticVersion(string);
+  }
+
+  obj->Wrap(args.This());
+  args.GetReturnValue().Set(args.This());
+}
+
+void SemanticVersion::Create(const FunctionCallbackInfo<Value>& args) {
+  Isolate *isolate = args.GetIsolate();
+  const unsigned argc = 5;
+  Local<Value> argv[argc] = { args[0], args[1], args[2], args[3], args[4] };
```

Comment:
I wonder if it just blindly accesses invalid memory (classic C[++] style). That might explain the test failure you see.